web/modules/custom/brms_common/src/DimensionCompute.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Drupal\brms_common;
+
+use Drupal\Core\Field\FieldItemList;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\TypedData\ComputedItemListTrait;
+use Drupal\geofield\GeoPHP\GeoPHPInterface;
+use Drupal\taxonomy\Entity\Term;
+
+/**
+ * Class DimensionCompute, to compute the values of the dimension fields.
+ *
+ * @package Drupal\brms_common
+ */
+class DimensionCompute extends FieldItemList implements FieldItemListInterface {
+
+  use ComputedItemListTrait;
+
+  /**
+   * Compute the values.
+   */
+  protected function computeValue() {
+    /** @var \Drupal\node\Entity\Node $node */
+    $node = $this->getParent()->getValue();
+    $dimension_name = $this->getFieldDefinition()['label'];
+    $dimension_name = (string) $dimension_name;
+    $dimension_id = $this->getName();
+    $dimension_settings = $node->get($dimension_id)->getSettings();
+    $dimension_to_calculate = $dimension_settings['computed_field'];
+    $dimension_computed_id = $dimension_settings['layer_type_id'];
```

Comment:
See above - we should not need this setting because we already have the $dimension_id